rfd/XXXX-access-list-templates.md

```diff
@@ -0,0 +1,328 @@
+
+##### System-managed roles for long-term template
+
+- access: a role that defines access to general resources - assigned to member grants
+- access-aws-ic: a role that defines access specific to AWS identity center - assigned to member grants
+
+##### Naming format for the system-managed roles
+
+In order to ensure uniqueness, the naming convention takes the following format:
+
+`templated-acl-<purpose>-role-<UID of access list>`
+
+| Parts                 |                   Explanation                    |                             Example Values |
+| :-------------------- | :----------------------------------------------: | -----------------------------------------: |
+| templated-acl         |        stands for "templated access list"        |                                        n/a |
+| \<purpose>-role       |  short word that describes the purpose of role   | requester, reviewer, access, access-aws-ic |
+| \<UID of access list> | describes which access list this role belongs to |                                        n/a |
+
+Example names of system-managed roles, if an access list with metadata.name is `b037d55d-c076-474e-b4a8-bc4d0f10f19d`
+
+- templated-acl-requester-role-b037d55d-c076-474e-b4a8-bc4d0f10f19d
+- templated-acl-reviewer-role-b037d55d-c076-474e-b4a8-bc4d0f10f19d
+- templated-acl-access-role-b037d55d-c076-474e-b4a8-bc4d0f10f19d
+- templated-acl-access-aws-ic-role-b037d55d-c076-474e-b4a8-bc4d0f10f19d
+
+#### Update
+
+`spec.type: templated`, `template_config.type`, member grant, and owner grant fields will not be modifiable.
+
+Only the `template_config.allow` is modifiable.
+
+Any updates made on a templated access list will upsert roles regardless if access has actually changed. This is in part to keep the roles in sync. When updating roles, the previous version of the role will be maintained to prevent unintended role behavioral changes with different role verisons. If a user wants to upgrade to a newer role version, user must create a new access list, which will use the latest role version.
+
+When modifying, Teleport will also delete orphaned roles if present. A role can become orphaned if a user changes access in a way it changes grants. E.g: if a user removes access for AWS IC, the role created specifically for AWS IC access becomes unused.
+
+#### Delete
+
+In the backend, after an access list is successfully deleted, all roles tied to that deleted access list will also be deleted. Note that the access list must be deleted first as roles assigned to access list cannot be deleted.
+
+##### Reconciler that cleans up orphaned roles
+
+If deleting roles fail for some reason, it leaves behind orphaned roles. A delete reconciler will be created where periodically the reconciler will iterate through all the roles, and if a role belonged to an access list and that access list no longer exists, reconciler will delete the role.
```

Comment:
i know we were against using any reconciler... but I didn't see a better way to deal with roles that got left behind

Reply:
Pretty sure we can do this without a reconciler as well. For example, delete the role(-s) first and only if that succeeds delete the list. This should not cause any issues even if list deletion fails for whatever reason because there was a change made a while ago that made lists gracefully degrade if they grant a non-existent role. But even if not, we can also always unassign the role from the list first (don't think it's necessary but should test).